### PR TITLE
oqs-provider: init at 0.6.1

### PR DESCRIPTION
**Pull request by Rixxc**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -17163,6 +17163,12 @@
     githubId = 449990;
     name = "Cedric Cellier";
   };
+  rixxc = {
```
Comment (sikmir) [inline maintainers/maintainer-list.nix:17166]: Please add yourself to maintainer list in separate commit `maintainers: add rixxc`

File: pkgs/by-name/oq/oqs-provider/package.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  preInstall = ''
+    mkdir -p "$out"
+    for dir in "$out" "${openssl.out}"; do
+      mkdir -p .install/"$(dirname -- "$dir")"
+      ln -s "$out" ".install/$dir"
+    done
+    export DESTDIR="$(realpath .install)"
+  '';
+
+  enableParallelBuilding = true;
```
Comment (sikmir) [inline pkgs/by-name/oq/oqs-provider/package.nix:46]:
```suggestion
```
It's enabled by default for cmake.

**Comment by Rixxc** [comment]
Hey, Thanks for the feedback. I've implemented your requests. Please let me know if there is anything else that needs to be changed.

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 326373` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>oqs-provider</li>
  </ul>
</details>

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 326373` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>oqs-provider</li>
  </ul>
</details>

File: pkgs/by-name/oq/oqs-provider/package.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  doCheck = true;
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = with lib; {
+    homepage = "https://github.com/open-quantum-safe/oqs-provider";
+    description = "Open Quantum Safe provider for OpenSSL (3.x)";
+    license = licenses.mit;
+    maintainers = with maintainers; [ rixxc ];
+    platforms = platforms.x86;
```
Comment (sikmir) [inline pkgs/by-name/oq/oqs-provider/package.nix:57]: Why x86 only? It builds for me on arm as well.

Comment (Rixxc) [reply pkgs/by-name/oq/oqs-provider/package.nix]: You're right with that. I thought liboqs is x86 only, but I was mistaken.

**Comment by Rixxc** [comment]
Hey, I think I've made all the changes you requested. Is there anything else I should do to get this PR merged?

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**